src/hono.test.ts

```diff
@@ -297,6 +297,10 @@ describe('Routing', () => {
     expect(res.status).toBe(200)
     expect(await res.text()).toBe('get /book/123')
 
+    res = await app.request('http://localhost/book/keys', { method: 'GET' })
```

Comment:
The problem occurs not only with `nested routes` but also in the following case:

```ts
const app = new Hono({ router: new TrieRouter() })
app.get('/foo/:id', (c) => c.text(c.req.param('id')))

// GET /foo/keys
```

So, you might not want to write the test in this section. I think a better place could be under `param and query`.

Reply:
Indeed!